### DISCUSSION TITLE
context.registration / context.extensions.(...)sessionid

### DISCUSSION THREAD

**Discussion by sfraysse** [gaia-x-dases/xapi-virtual-classroom#5] [General]
I would suggest to add a `context.registration` prop to all the templates of this profile.

This would be consistent with other well known profiles such as xAPI video profile or CMI5 profile.
And this may be usefull during analytics when you want to group consecutive statements.
 
But the use of this property may be a bit confusing with some profiles so we should be clear about its meaning. 

Let's take a common scenario and define some terms (just for a better understanding):

- In the FUN training catalogue, there is a course named "xAPI 101", which is proposed in a virtual classroom format. **< COURSE**
- There is a session scheduled the 10th Jan 2023 from 2:00pm to 3:00pm. **< SCHEDULED SESSION**
- The instructor starts the session on time. **< 1st TECHNICAL SESSION**
- At 2:13pm, there is a technical issue and the instructor has to relaunch the session. **< 2ND TECHNICAL SESSION**

From my point of view, the `context.registration` prop should identify what I called the **SCHEDULED SESSION**. This is the session learners have registred to. So the term **registration** is consistent here.

We may add a context extension to identify what I called the **TECHNICAL SESSION**, which is a number or UUID generated for the `initialized` statement, and which stays the same until the `terminated` statement. When the instructor relaunches the virtual classroom, there is a new `initialized` statement, and so a new `registration`.

So to summarize these 2 first points, we could have something like this, which is very consistent with the CMI5 profile:
```
"context": {
    "registration": "4eb0e063-669b-479a-86b3-f9be9ac88a1d",
    "extensions": {
        "https://w3id.org/xapi/cmi5/context/extensions/sessionid": "c7b6f0a9-482c-4c03-acc1-548289126963"
    }
}
```

Finally, what I called a **COURSE** (don't be disturbed by this term) would be identified by an activity in the `object` of the statements (sometimes in the `contextActivities.parent`). Something like:
```
{
    "objectType": "Activity",
    "id": "http://gaia-x-dases.org/courses/xapi-101",
    "definition": {
        "type": "http://id.tincanapi.com/activitytype/webinar"
    }
}
```

**Comment by quitterie-lcs** [comment]
I was not definitely not aware of this information to be contained in this field. Indeed it makes really sense to track this information in this way.

**Comment by sfraysse** [comment]
Sorry, just reading again my post, I find a small mistake:

> When the instructor relaunches the virtual classroom, there is a new initialized statement, and so a new registration.

I wanted to say **"and so a new session id"**. The registration should stay the same if it is the same planned session.

**Comment by quitterie-lcs** [comment]
For the session ID, why do you choose the CMI5 definition rather than the one defined in the video profile ? 
I raise this question as in the video profile,it does not specify a course registration and would be more generic: 
- [Video session ID](https://profiles.adlnet.gov/profile/0bfffef6-b5f1-4aad-8fdf-9253d7cd42f9/concepts/e268c08f-b488-4370-a4b0-37afcaf60e96)
- [CMI5 session ID](https://profiles.adlnet.gov/profile/a929b474-9518-45a2-bd47-24696c602754/concepts/bb4869fe-560d-4b89-a54b-2741761e37c4)

**Reply by sfraysse** [reply]
You are right, the CMI5 description on the profile server refers to the registration, and not video profile description.

However, refering to the human-readable specs, these 2 concepts exist and have the same meaning, and the video profile spec also refers to the registration concept, working together with the session concept:

> A session starts with the initiated statement and ends with terminated statement.

> The registration value is important because the "progress", the "played-segments" value and "completion" is based on aggregate of all attempts with same registration value attempted by the Actor for the specific media.

An **attempt** is the same thing as a **session**.

I took the CMI5 session IRI only because CMI5 seams to be considered as a reference profile by ADL. For example, CMI5 is referred in the MOM profile:

> Shall conform to the cmi5 Specification (in regard to xAPI statements)

So I tried to stay as close as possible from the CMI5 profile.

But yes, this is disturbing to see several profiles refering to the same concept with a different IRI, and a more "generic" IRI would probably be better. This is a question for ADL.

That's something to keep in mind if this spec is rewied by ADL in the future...
